api/backend.go

```diff
@@ -555,6 +558,53 @@ func (b *StatusBackend) GetNodesFromContract(rpcEndpoint string, contractAddress
 	return response, nil
 }
 
+// VerifyENSName verifies that a registered ENS name matches the expected public key
+func (b *StatusBackend) VerifyENSName(ensName, publicKeyStr, rpcEndpoint, contractAddress string) (bool, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), contractQueryTimeout)
+	defer cancel()
+
+	pubKeyBytes, err := hex.DecodeString(publicKeyStr)
+	if err != nil {
+		return false, err
+	}
+
+	expectedPubKey, err := btcec.ParsePubKey(pubKeyBytes, btcec.S256())
```

Comment:
The same can be done with `crypto.UnmarshalPubkey()` where `crypto` is `github.com/ethereum/go-ethereum/crypto`.